securitycenter/notifications/receive_notifications.go

```diff
@@ -0,0 +1,45 @@
+package notifications
+
+// [START scc_receive_notifications]
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/golang/protobuf/jsonpb"
+	securitycenterpb "google.golang.org/genproto/googleapis/cloud/securitycenter/v1p1beta1"
+)
+
+func parseNotificationMessage(ctx context.Context, msg *pubsub.Message) {
+	var notificationMessage = new(securitycenterpb.NotificationMessage)
+	u := &jsonpb.Unmarshaler{}
+	u.Unmarshal(bytes.NewReader(msg.Data), notificationMessage)
+
+	fmt.Printf("Got finding: %v", notificationMessage.GetFinding())
+	msg.Ack()
+}
+
+func receiveMessages(projectId string, subscriptionName string) error {
+	ctx := context.Background()
+
+	// TODO projectId := "your-project-id"
+	// TODO subsriptionName := "your-subscription-name"
+
+	client, err := pubsub.NewClient(ctx, projectId)
+	if err != nil {
+		return fmt.Errorf("pubsub.NewClient: %v", err)
+	}
+
+	sub := client.Subscription(subscriptionName)
+	cctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	err = sub.Receive(cctx, parseNotificationMessage)
```

Comment:
> Receive calls f with the outstanding messages from the subscription. It blocks until ctx is done, or the service returns a non-retryable error.

https://pkg.go.dev/cloud.google.com/go/pubsub?tab=doc#Subscription.Receive

Should `cctx` be canceled after one message?

Reply:
I do want the function to cancel after receiving a message for the sake of tests. I believe I've fixed this, but if not will need some guidance.